db/memtable.cc

```diff
@@ -575,28 +811,36 @@ Status MemTable::Add(SequenceNumber s, ValueType type,
 
   size_t ts_sz = GetInternalKeyComparator().user_comparator()->timestamp_size();
   Slice key_without_ts = StripTimestampFromUserKey(key, ts_sz);
-
+  uint32_t inserted_entries;
+  uint32_t inserted_len;
   if (!allow_concurrent) {
     // Extract prefix for insert with hint.
     if (insert_with_hint_prefix_extractor_ != nullptr &&
-        insert_with_hint_prefix_extractor_->InDomain(key_slice)) {
-      Slice prefix = insert_with_hint_prefix_extractor_->Transform(key_slice);
-      bool res = table->InsertKeyWithHint(handle, &insert_hints_[prefix]);
+        insert_with_hint_prefix_extractor_->InDomain(key)) {
+      Slice prefix = insert_with_hint_prefix_extractor_->Transform(key);
+      bool res = InsertKey(table, handle, s, type, key, value, encoded_len,
+                           &inserted_entries,
+                           &inserted_len,
+                           std::bind(&MemTableRep::InsertKeyWithHint,
+                           std::placeholders::_1,
+                           std::placeholders::_2,
+                           &insert_hints_[prefix]));
```

Comment:
ASAN seems to complain about this line. I see `prefix` here is built from`key`, but [before this PR](https://github.com/facebook/rocksdb/blob/1e9bf25f61fa5b08fe2b3fe5c1fcf33e4c1c515c/db/memtable.cc#L583) it's built from `key_slice` which is based on the keyhandle.